tests/primitives/test_tool.py

```diff
@@ -159,3 +159,17 @@ def foo(x=100):
     tool = Tool(foo)
     assert tool.args["x"]["default"] == 100
     assert not hasattr(tool.args["x"], "type")
+
+
+def test_tool_call_parses_args():
```

Comment:
nit: Can we also add a unit test for the nested type like `list[DummyModel]`

Reply:
good call!